### ISSUE TITLE
inceptionnext performance

### ISSUE THREAD

**Issue by suisui-su-crtl** [sail-sg/inceptionnext#10]
Hello, I tried to use inceptionnext as the encoder, but the experiment results are not ideal ( The performance is even worse than convnext, I think maybe I made a mistake), can you help me to see if there is a problem with my code?

Since the encoder is connected to the decoder at each layer, I need to extract the results of each layer.

original:

    def forward(self, x):
        x = self.forward_features(x)
        x = self.forward_head(x)  # I removed this part of the code because of the encoder
        return x

I modified:

    def forward(self, x):
        y = x
        y = self.stem(y)
        y = self.stages[0] (y)
        x1 = y
        
        y = self.stages[1] (y)
        x2 = y
        
        y = self.stages[2] (y)
        x3 = y
        
        y = self.stages[3] (y)
        x4 = y
        
        return x1, x2, x3, x4

**Comment by yuweihao** [comment]
Hi, for the semantic segmentation task in the paper, we add SyncBN for the output of each stage, like
`return syncbn1(x1), syncbn2(x2), syncbn3(x3), syncbn4(x4)`.

**Comment by suisui-su-crtl** [comment]
> 
Thank you very much for your help. My research area is semantic segmentation.

Do you mean to add SyncBN layer after the output of each layer?

I have some questions about SyncBN layer.

I don't see the use of SyncBN in code. Do I need to add it by myself?

I use a single card for training, search found that SyncBN can support multi-card training, syncbn is useful for my training? Or is it more appropriate to directly add a BN layer? 

Looking forward to your reply, which is very important to me.

**Comment by yuweihao** [comment]
Hi, yes, BN needs to add for the semantic segmentation task. For a single GPU, BN is equal to SyncBN. For multi-GPU training, please remember to use SyncBN.

**Comment by 116022017144** [comment]
in instance segmentation task,I didn't use the BN for the output of each stage，and my result of experiment is bad.

**Comment by 116022017144** [comment]
I don't know the reason.I use inceptionnext-tiny to replace resnet50, why the result is worse.

**Comment by yuweihao** [comment]
Hi @116022017144 ,

Besides adding SyncBN for the output of each stage, remember to replace BN in InceptionNeXt with SyncBN.

**Comment by 116022017144** [comment]
ok,thank you！Do all the BN is needed to be replaced?

**Comment by 116022017144** [comment]
还有对每个stage输出做SyncBN，这样可以吗？self.stages = nn.Sequential(norm_layer(dims[0]))，还是只需要在声明形参那里这样 norm_layer=nn.SyncBatchNorm,就行

**Comment by 116022017144** [comment]
这个类class MlpHead(nn.Module)，如果做分割任务的话，我看是没有用到的，其norm层不用管了吧？

**Comment by yuweihao** [comment]
Hi @116022017144 , yes, I remove all MlpHead for semantic segmentation.

**Comment by 116022017144** [comment]
![image](https://github.com/sail-sg/inceptionnext/assets/33896598/ccf2c064-cd3d-4836-85d4-8a3ae8f9ef9b)
I met the error. And my model support distrbuted training.

**Comment by suisui-su-crtl** [comment]
> Hi @116022017144 , yes, I remove all MlpHead for semantic segmentation.

Can you come up with a semantic segmentation version of inceptionnext when you have time? I would really appreciate it if you could!

I added BN layers after each output layer, but the accuracy is only slightly improved, similar to the performance of convnext. I'm having a hard time finding the problem.

**Comment by 116022017144** [comment]
sorry,I use it in instance segmentation task, so I can't help you! Now I don't replace BN，because I met some errors



---Original---
From: ***@***.***&gt;
Date: Thu, May 25, 2023 22:03 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [sail-sg/inceptionnext] inceptionnext performance (Issue #10)

**Comment by yuweihao** [comment]
> ![image](https://user-images.githubusercontent.com/33896598/240611745-ccf2c064-cd3d-4836-85d4-8a3ae8f9ef9b.png) I met the error. And my model support distrbuted training.

Hi, I use MMSeg for semantic segmentation. SyncBN code like this
```python
from mmcv.cnn import build_norm_layer

norm_cfg=dict(type='SyncBN', requires_grad=True)

self.norm = build_norm_layer(norm_cfg, out_chs)[1]
```

**Comment by yuweihao** [comment]
> > Hi @116022017144 , yes, I remove all MlpHead for semantic segmentation.
> 
> Can you come up with a semantic segmentation version of inceptionnext when you have time? I would really appreciate it if you could!
> 
> I added BN layers after each output layer, but the accuracy is only slightly improved, similar to the performance of convnext. I'm having a hard time finding the problem.

Hi, sorry that I am so busy recently and may have no time to clean the code. Following ConvNeXt, we report mIoU multi-scale testing in Table 5 in the paper, where InceptionNeXt significantly outperforms ConvNeXt. Besides, I also evaluate it with single scale, finding the performance of InceptionNeXt is similar to ConvNeXt. It seems InceptionNeXt is more in favor of multi-scale testing.

**Comment by suisui-su-crtl** [comment]
> > > Hi @116022017144 , yes, I remove all MlpHead for semantic segmentation.
> > 
> > 
> > Can you come up with a semantic segmentation version of inceptionnext when you have time? I would really appreciate it if you could!
> > I added BN layers after each output layer, but the accuracy is only slightly improved, similar to the performance of convnext. I'm having a hard time finding the problem.
> 
> Hi, sorry that I am so busy recently and may have no time to clean the code. Following ConvNeXt, we report mIoU multi-scale testing in Table 5 in the paper, where InceptionNeXt significantly outperforms ConvNeXt. Besides, I also evaluate it with single scale, finding the performance of InceptionNeXt is similar to ConvNeXt. It seems InceptionNeXt is more in favor of multi-scale testing.

Ok, I can understand. Thank you very much for your help many times. I will read the paper in detail again to find the answer.

**Comment by iz626** [comment]
> > > > Hi [@116022017144](https://github.com/116022017144) , yes, I remove all MlpHead for semantic segmentation.
> > > 
> > > 
> > > Can you come up with a semantic segmentation version of inceptionnext when you have time? I would really appreciate it if you could!
> > > I added BN layers after each output layer, but the accuracy is only slightly improved, similar to the performance of convnext. I'm having a hard time finding the problem.
> > 
> > 
> > Hi, sorry that I am so busy recently and may have no time to clean the code. Following ConvNeXt, we report mIoU multi-scale testing in Table 5 in the paper, where InceptionNeXt significantly outperforms ConvNeXt. Besides, I also evaluate it with single scale, finding the performance of InceptionNeXt is similar to ConvNeXt. It seems InceptionNeXt is more in favor of multi-scale testing.
> 
> Ok, I can understand. Thank you very much for your help many times. I will read the paper in detail again to find the answer.

Hello,I noticed that you previously encountered an issue while using InceptionNeXt as an encoder backbone for a semantic segmentation network. I’m currently facing a similar challenge and wanted to check if you were able to resolve it. If so, could you kindly share any insights or reference code that helped you address the problem? Your guidance would be immensely valuable to my research.  Thank you for your time, and I look forward to your response!  
你好，我注意到你之前在使用InceptionNeXt作为语义分割网络的编码器主干时遇到了问题。目前我也遇到了类似的挑战，想确认你是否已成功解决该问题？如果是的话，能否分享一些解决思路或参考代码？你的建议对我的研究将非常有帮助！感谢你的时间，期待你的回复！

**Comment by 116022017144** [comment]
不好意思，我现在没做这个研究了，之前的实验数据也删了，不能帮助你哦



---Original---
From: ***@***.***&gt;
Date: Tue, Mar 4, 2025 15:47 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [sail-sg/inceptionnext] inceptionnext performance (Issue #10)


       
Hi @116022017144 , yes, I remove all MlpHead for semantic segmentation.
  
Can you come up with a semantic segmentation version of inceptionnext when you have time? I would really appreciate it if you could!
 I added BN layers after each output layer, but the accuracy is only slightly improved, similar to the performance of convnext. I'm having a hard time finding the problem.
  
Hi, sorry that I am so busy recently and may have no time to clean the code. Following ConvNeXt, we report mIoU multi-scale testing in Table 5 in the paper, where InceptionNeXt significantly outperforms ConvNeXt. Besides, I also evaluate it with single scale, finding the performance of InceptionNeXt is similar to ConvNeXt. It seems InceptionNeXt is more in favor of multi-scale testing.
  
Ok, I can understand. Thank you very much for your help many times. I will read the paper in detail again to find the answer.
 Hello,I noticed that you previously encountered an issue while using InceptionNeXt as an encoder backbone for a semantic segmentation network. I’m currently facing a similar challenge and wanted to check if you were able to resolve it. If so, could you kindly share any insights or reference code that helped you address the problem? Your guidance would be immensely valuable to my research.  Thank you for your time, and I look forward to your response!
 你好，我注意到你之前在使用InceptionNeXt作为语义分割网络的编码器主干时遇到了问题。目前我也遇到了类似的挑战，想确认你是否已成功解决该问题？如果是的话，能否分享一些解决思路或参考代码？你的建议对我的研究将非常有帮助！感谢你的时间，期待你的回复！
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***&gt;
  iz626 left a comment (sail-sg/inceptionnext#10)
     
Hi @116022017144 , yes, I remove all MlpHead for semantic segmentation.
  
Can you come up with a semantic segmentation version of inceptionnext when you have time? I would really appreciate it if you could!
 I added BN layers after each output layer, but the accuracy is only slightly improved, similar to the performance of convnext. I'm having a hard time finding the problem.
  
Hi, sorry that I am so busy recently and may have no time to clean the code. Following ConvNeXt, we report mIoU multi-scale testing in Table 5 in the paper, where InceptionNeXt significantly outperforms ConvNeXt. Besides, I also evaluate it with single scale, finding the performance of InceptionNeXt is similar to ConvNeXt. It seems InceptionNeXt is more in favor of multi-scale testing.
  
Ok, I can understand. Thank you very much for your help many times. I will read the paper in detail again to find the answer.
 Hello,I noticed that you previously encountered an issue while using InceptionNeXt as an encoder backbone for a semantic segmentation network. I’m currently facing a similar challenge and wanted to check if you were able to resolve it. If so, could you kindly share any insights or reference code that helped you address the problem? Your guidance would be immensely valuable to my research.  Thank you for your time, and I look forward to your response!
 你好，我注意到你之前在使用InceptionNeXt作为语义分割网络的编码器主干时遇到了问题。目前我也遇到了类似的挑战，想确认你是否已成功解决该问题？如果是的话，能否分享一些解决思路或参考代码？你的建议对我的研究将非常有帮助！感谢你的时间，期待你的回复！

**Comment by iz626** [comment]
> 不好意思，我现在没做这个研究了，之前的实验数据也删了，不能帮助你哦
> […](#)

Thank you for your response! Wishing you all the best with your current and future projects!
